code/obj/machinery/law_racks.dm

```diff
@@ -0,0 +1,461 @@
+				welded[slotNum] ? "You unscrew the module." : "You screw the module into the rack.", \
+				INTERRUPT_ACTION | INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ACT)
+
+				return
+			if("rack")
+				if (welded[slotNum])
+					ui.user.visible_message("<span class='alert'>[ui.user] tries to tug a module out of the rack, but it's welded in place!</span>", "<span class='alert'>You struggle with the module but it's welded in place!</span>")
+					return
+				if (screwed[slotNum])
+					ui.user.visible_message("<span class='alert'>[ui.user] tries to tug a module out of the rack, but it's still screwed in!</span>", "<span class='alert'>You struggle with the module but it's still screwed in!</span>")
+					return
+
+				if(law_circuits[slotNum])
+					//add circuit to hand
+					ui.user.visible_message("<span class='alert'>[ui.user] slides a module out of the law rack</span>", "<span class='alert'>You slide the module out of the rack.</span>")
+					ui.user.put_in_hand_or_drop(law_circuits[slotNum])
+					law_circuits[slotNum] = null
+					UpdateIcon()
+					UpdateLaws()
+				else
+
+					var/equipped = ui.user.equipped()
+					if(!equipped)
+						return
+
+					if(!istype(equipped,/obj/item/aiModule))
+						ui.user.visible_message("<span class='alert'>[ui.user] tries to shove \a [equipped] into the rack. Silly [ui.user]!</span>", "<span class='alert'>You try to put \a [equipped] into the rack. You feel very foolish.</span>")
+						return
+
+					law_circuits[slotNum]=equipped
+					ui.user.u_equip(equipped)
+					ui.user.visible_message("<span class='alert'>[ui.user] slides a module into the law rack</span>", "<span class='alert'>You slide the module into the rack.</span>")
+
+					UpdateIcon()
+					UpdateLaws()
+
+	proc/show_laws(var/who)
+		var/list/L = who
+		if (!istype(who, /list))
+			L = list(who)
+
+		var/laws_text = src.format_for_logs()
+		for (var/W in L)
+			boutput(W, laws_text)
+
+	proc/format_for_logs(var/glue = "<br>")
+		var/law_counter = 1
+		var/lawOut = list()
+		for (var/obj/item/aiModule/X in law_circuits)
+			if(!X)
+				continue
+			lawOut += "[law_counter++]: [X.get_law_text()]"
+
+		return jointext(lawOut, glue)
+
+	proc/format_for_irc()
+		var/list/laws = list()
+
+		var/law_counter = 1
+		for (var/obj/item/aiModule/X in law_circuits)
+			if(!X)
+				continue
+			laws["[law_counter]"] = X.get_law_text()
+			law_counter++
+
+		return laws
+
+	proc/UpdateLaws(var/notification_text="<h3>Law update detected</h3>")
+		for (var/mob/living/silicon/R in mobs)
+			if (isghostdrone(R))
+				continue
+			if(R.law_rack_connection == src)
+				R << sound('sound/misc/lawnotify.ogg', volume=100, wait=0)
+				R.show_text(notification_text, "red")
+			src.show_laws(R)
+
+		for (var/mob/living/intangible/aieye/E in mobs)
+			if(E.mainframe?.law_rack_connection == src)
+				E << sound('sound/misc/lawnotify.ogg', volume=100, wait=0)
```

Comment:
You'll want playsound_player or_local or whatever it is ask imcoder

Applies for all these s << sounds